server/liveblog/syndication/syndication.py

```diff
@@ -174,14 +180,25 @@ def syndication_webhook():
     new_post = create_syndicated_blog_post(producer_post, items, in_syndication)
     posts_service = get_resource_service('posts')
     producer_post_id = new_post['producer_post_id']
-    post = posts_service.find_one(req=None, producer_post_id=producer_post_id)
-    # TODO: update post
-    if not post:
+
+    method = request.method
+
+    if method == 'POST':
         new_post_id = posts_service.post([new_post])[0]
         return api_response({'post_id': str(new_post_id)}, 201)
     else:
-        logger.warning('Producer post "{}" already exists'.format(producer_post_id))
-        return api_error('Producer post already exists!', 409)
+        post = posts_service.find_one(req=None, producer_post_id=producer_post_id)
+        if not post:
+            return api_error('Post cannot be updated: syndicated post does not exist.', 404)
+
+        post_id = str(post['_id'])
+        publisher = post.get('publisher')
+        if not publisher and method == 'PUT':
+            # Update only posts with publisher set to None
+            posts_service.put(post['_id'], new_post)
+            return api_response({'post_id': post_id}, 200)
+        else:
+            return api_error('Post "{}" cannot be updated: publisher already exists.'.format(post_id), 409)
```

Comment:
No sure that this should return an error, as it is an expected case and the source of the post doesn't have to respond to it